SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -364,29 +368,41 @@ - (void)didEnterStateRegistered {
         [self sdl_transitionToState:SDLLifecycleStateUnregistering];
         return;
     }
-
     NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
-    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
-    SDLLanguage actualLanguage = self.registerResponse.language;
-    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:)];
-    
+    // the language represents hmi language
+    SDLLanguage desiredHmiLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage desiredVrLanguage = self.vrLanguage;;
+
+    SDLLanguage actualHmiLanguage = self.registerResponse.hmiDisplayLanguage;
+    SDLLanguage actualVrLanguage = self.registerResponse.language;
+
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:hmiLanguage:)];
```

Comment:
You can't just check the new one, you have to check if they implement _either_ the new method or the older method.